src/BlockTypes/ProductSaleBadge.php

```diff
@@ -63,6 +63,68 @@ protected function get_block_type_supports() {
 		);
 	}
 
+	/**
+	 * Initialize this block.
+	 */
+	protected function initialize() {
+		parent::initialize();
+		add_filter(
+			'wp_theme_json_data_theme',
+			array( $this, 'add_default_sale_badge_styles' )
+		);
+	}
+
+	/**
+	 * Add default styles for Sale Badge block.
+	 *
+	 * @param WP_Theme_JSON_Data $theme_data The theme data object that is passed as a parameter.
+	 * @return array
+	 */
+	public function add_default_sale_badge_styles( $theme_data ) {
+		// Get the existing Sale Badge block styles.
+		$block_styles = isset( $theme_data->get_data()['styles']['blocks']['woocommerce/product-sale-badge'] )
+		? $theme_data->get_data()['styles']['blocks']['woocommerce/product-sale-badge']
+		: array();
+
+		// Return if there are no styles for button element.
+		// Because in this case Sale badge will use styles from style.scss.
+		$button_element = $theme_data->get_data()['styles']['elements']['button'] ?? null;
+		if ( ! $button_element ) {
+			return $theme_data;
+		}
+
+		$default_styles = array(
+			'version' => 2,
+			'styles'  => array(
+				'blocks' => array(
+					'woocommerce/product-sale-badge' => array(
+						'color' => array(),
+					),
+				),
+			),
+		);
+
+		if ( ! isset( $block_styles['color'] ) || ! isset( $block_styles['color']['background'] ) ) {
```

Comment:
What do you think about simplifying this condition to:

```suggestion
		if ( ! isset( $block_styles['color']['background'] ) ) {
```

Reply:
Oh, I see. Thank you for letting me know. In some programming languages, attempting to access a property that doesn't exist can result in an exception or error. However, it seems like that's not the case in PHP. In PHP, if you try to access a property that doesn't exist, PHP will just return null. So, the check for the existence of the 'color' key in the original code is redundant in this case, since if the key doesn't exist, the condition will evaluate to false anyway. I appreciate your feedback and the opportunity to learn more about PHP. 🙌🏻